homework_5/numpy_challenge.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+
+
+def matrix_multiplication(m1, m2):
+    result = []
+    nrow_m1 = m1.shape[0]
+    ncol_m2 = m2.shape[1]
+    for i in range(nrow_m1):
+        for j in range(ncol_m2):
+            dot = m1[i, :] * m2[:, j]
+            result.append(dot.sum())
+    result = np.array(result).reshape(nrow_m1, ncol_m2)
+    return result
+
+
+def multiplication_check(matrix_list: list) -> bool:
+    ncol = matrix_list[0].shape[1]
+    for i in range(1, len(matrix_list)):
+        nrow = matrix_list[i].shape[0]
+        if nrow == ncol:
+            ncol = matrix_list[i].shape[1]
+        else:
+            return False
+    return True
+
+
+def multiply_matrices(matrix_list: list):
+    if multiplication_check(matrix_list) is False:
+        return None
+    step = matrix_multiplication(matrix_list[0], matrix_list[1])
+    for i in range(2, len(matrix_list)):
+        step = matrix_multiplication(step, matrix_list[i])
+    return step
+
+
+def compute_multidimensional_distance(arr1, arr2):
+    coord = arr2 - arr1
+    coord_sq = coord ** 2
+    dist_sq = coord_sq.sum()
+    return dist_sq ** 0.5
+
+
+def compute_2d_distance(arr1, arr2):
+    return compute_multidimensional_distance(arr1, arr2)
+
+
+def compute_pair_distances(matrix):
+    nrow = matrix.shape[0]
+    dist_matrix = np.zeros(nrow ** 2).reshape(nrow, nrow)
```

Comment:
```suggestion
    dist_matrix = np.zeros((nrow, nrow))
```
Зачем решейпить, если можно создать сразу заданного размера